x/bank/types/send_authorization.go

```diff
@@ -6,13 +6,26 @@ import (
 	"github.com/cosmos/cosmos-sdk/x/authz"
 )
 
+// TODO: Revisit this once we have propoer gas fee framework.
+// Tracking issues https://github.com/cosmos/cosmos-sdk/issues/9054, https://github.com/cosmos/cosmos-sdk/discussions/9072
+const gasCostPerIteration = uint64(10)
+
 var _ authz.Authorization = &SendAuthorization{}
 
 // NewSendAuthorization creates a new SendAuthorization object.
-func NewSendAuthorization(spendLimit sdk.Coins) *SendAuthorization {
-	return &SendAuthorization{
-		SpendLimit: spendLimit,
+func NewSendAuthorization(allowed []sdk.AccAddress, spendLimit sdk.Coins) (*SendAuthorization, error) {
+	allowedAddrs := toBech32Addresses(allowed)
+
+	a := SendAuthorization{}
+	if allowedAddrs != nil {
+		a.AllowList = allowedAddrs
+	}
```

Comment:
```suggestion
	a.AllowList = allowedAddrs
```

I think this is equivalent, and shorter